pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1234,6 +1247,8 @@ func encodeNetworkPortConfig(npc *types.NetworkPortConfig) *info.DevicePort {
 		errInfo.Timestamp = errTime
 		dp.Err = errInfo
 	}
+
+	dp.NetworkID = npc.NetworkUUID.String()
```

Comment:
This doesn't match the comment, because you set it to a zero UUID when it is not a management port.
Check if the uuid is zero before setting?
And update the comment in the proto file because you'd still be setting it for any ports which have a network, e.g., app-shared.

Reply:
I made it to set NetworkUUID only when it is not a nil value and also updated the comment..